Add ui segmented control

nfarah86 opened the pull request:
@gwalker1228 please review!  Thanks!

nfarah86 commented:
@mplewis please review :)

nfarah86 commented on CardMatchingGameWithBean/Model/CardMatchingGame.m:
```diff
@@ -45,48 +49,93 @@ -(instancetype) initWithCardCount:(NSUInteger)count usingDeck:(Deck *)deck
 
+- (NSInteger)_differentiateTwoAndThreeCardGames:(NSInteger)segmentControlIndex
+{
+    return [self.chosenCards count] == segmentControlIndex + 2 ? [self.chosenCards count]: 0;
```
In order for me to set the segmentControlIndex properly, I'll have to add the cards in the array in the ViewController, and put the matching logic there (how cards are being flipped as they pick it.).  As the user are picking cards, the cards are flipping over.  If I add the cards in an array in the ViewController, then handle the flipping logic in this current file, the cards won't flip when they pick it. It'll flip afterwards.

mplewis commented on CardMatchingGameWithBean/Model/CardMatchingGame.m:
```diff
@@ -45,48 +49,92 @@ -(instancetype) initWithCardCount:(NSUInteger)count usingDeck:(Deck *)deck
+                if (self.matchScore) {
+                    [self matchCards:self.chosenCards];
+                } else if (self.matchScore == 0) {
+                    [self unmatchCards:self.chosenCards];
```
`0` will automatically evaluate to `NO` in an if statement, so you don't have to use else if == 0, you can just use else

nfarah86 commented:
@mplewis I added a label that describes which cards are being flipped; and implemented a delegate protocol.  I used a delegate protocol to send that cards that are picked by the user to the view controller.  Then I write to the label which cards the user picked, and whether they have a match or not.
